crates/edr_napi/index.d.ts

```diff
@@ -1134,6 +1162,16 @@ export declare class EdrContext {
   createProvider(chainType: string, providerConfig: ProviderConfig, loggerConfig: LoggerConfig, subscriptionConfig: SubscriptionConfig, tracingConfig: TracingConfigWithBuffers): Promise<Provider>
   /**Registers a new provider factory for the provided chain type. */
   registerProviderFactory(chainType: string, factory: ProviderFactory): Promise<void>
+  registerSolidityTestRunnerFactory(chainType: string, factory: SolidityTestRunnerFactory): Promise<void>
+  /**
+   * Executes Solidity tests.
+   *
+   * The function will return as soon as test execution is started.
+   * The progress callback will be called with the results of each test
+   * suite. It is up to the caller to track how many times the callback
+   * is called to know when all tests are done.
+   */
+  runSolidityTests(chainType: string, artifacts: Array<Artifact>, testSuites: Array<ArtifactId>, configArgs: SolidityTestRunnerConfigArgs, tracingConfig: TracingConfigWithBuffers, onTestSuiteCompletedCallback: (result: SuiteResult) => void): Promise<void>
```

Comment:
I recall that we originally went with the `errorCallback` API to conform to proper JS standards. As we'll need to support a result type in the near future for gas reports, I've taken this shortcut here, where the API returns a promise upon completion of the factory construction.